python/ray/serve/utils.py

```diff
@@ -173,3 +175,36 @@ def unwrap_callback(fut: asyncio.Future):
     src.add_done_callback(unwrap_callback)
 
     return dest_futures
+
+
+class MockScheduler:
```

Comment:
Is there any reason the interface needs to be stateful and maintain the available resources? I think it would be simpler if you just made this a function that takes in the available resources and a list of resource dicts and returns `True`/`False`. No need to make this more complicated than it needs to be.